docs(divider): update documentation (#5629)
adobe/spectrum-web-components: packages/divider/README.md

[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/divider?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/divider)
[![Try it on Stackblitz](https://img.shields.io/badge/Try%20it%20on-Stackblitz-blue?style=for-the-badge)](https://stackblitz.com/edit/vitejs-vite-rqfjtpgz)

```zsh
yarn add @spectrum-web-components/divider
```

Import the side effectful registration of `<sp-divider>` via:

```js
import '@spectrum-web-components/divider/sp-divider.js';
```

When looking to leverage the `Divider` base class as a type and/or for extension purposes, do so via:

```js
import { Divider } from '@spectrum-web-components/divider';
```

> ### Options

Author: Yeah, there's no anatomy section in the design docs for this one

#### Horizontal

> By default, dividers are horizontal and should be used for separating content vertically.

Reviewer: ```suggestion
Horizontal dividers are used to separate content stacked vertically.
```
